packages/tenderly-core/src/internal/core/services/TenderlyService.ts

```diff
@@ -199,6 +201,134 @@ export class TenderlyService {
+      tenderlyApi = TenderlyApiService.configureInstance();
+    }
+
+    try {
+      if (request.contracts.length === 0) {
+        logger.error(NO_VERIFIABLE_CONTRACTS_ERR_MSG);
+        return;
+      }
+
+      const res = await tenderlyApi.post("/api/v1/public/contracts/verify", { ...request });
+      if (res.data === undefined || res.data === null) {
+        logger.error(
+          "There was an error while publicly verifying contracts on Tenderly. Obtained response is invalid."
+        );
+        return;
+      }
+      const response = convertToLogCompliantVerificationResponse(res.data);
+      logger.trace("Verification response:", response);
+
+      if (response.compilation_errors !== undefined && response.compilation_errors !== null) {
+        logger.error("There have been compilation errors while verifying contracts.", response.compilation_errors);
+        return;
+      }
+
+      if (response.results === undefined || response.results === null) {
+        logger.error(
+          "There has been an error while verifying contracts, no verified contracts nor bytecode mismatch errors are returned."
+        );
+        return;
+      }
+
+      if (
+        response.results.bytecode_mismatch_errors !== undefined &&
+        response.results.bytecode_mismatch_errors !== null
+      ) {
+        for (const bytecodeMismatchError of response.results.bytecode_mismatch_errors) {
+          logger.error("There has been a bytecode mismatch error while verifying contract.", bytecodeMismatchError);
+        }
+      }
+      if (response.results.verified_contracts !== undefined && response.results.verified_contracts !== null) {
+        for (const verifiedContract of response.results.verified_contracts) {
+          const contractLink = `${TENDERLY_DASHBOARD_BASE_URL}/contract/${
+            CHAIN_ID_NETWORK_NAME_MAP[verifiedContract.network_id]
+          }/${verifiedContract.address}`;
+          const logMsg = `Contract ${verifiedContract.address} verified. You can view the contract at ${contractLink}`;
+          console.log(logMsg);
+          logger.trace(logMsg);
+        }
+      }
+    } catch (err) {
+      const logCompliantApiError = convertToLogCompliantApiError(err);
+      logger.error(logCompliantApiError);
+      logger.error(`Error in ${this.pluginName}: ${API_VERIFICATION_REQUEST_ERR_MSG}`);
+    }
+  }
+
+  public async pushContractsMultiCompiler(
+    request: TenderlyVerifyContractsRequest,
+    tenderlyProject: string,
+    username: string
+  ): Promise<void> {
+    logger.debug("Privately verifying contracts on tenderly. (Multi compiler version)");
+    if (!TenderlyApiService.isAuthenticated()) {
+      logger.error(`Error in ${this.pluginName}: ${ACCESS_TOKEN_NOT_PROVIDED_ERR_MSG}`);
+      return;
+    }
+
+    const tenderlyApi = TenderlyApiService.configureInstance();
+    try {
+      if (request.contracts.length === 0) {
+        logger.error(NO_VERIFIABLE_CONTRACTS_ERR_MSG);
+        return;
+      }
```

Comment:
Error handling seems pretty similar to the ones in other verification options. Maybe we can see if we can keep not duplicate the code, really important for errors.